### PR TITLE
Cvs 107522 holistic demo

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

**Comment by dtrawins** [comment]
do not add big images to github
use in the demo only an image with a single person which use all models (face,hand and pose)
simplify preparing the config and models - process should be fully automated without manual commands in the container, try to avoid building medipipe image as this is not fully relevant for ovms demo

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        if not os.path.exists(line.strip()):
+            print("Image does not exist: " + line.strip())
+        im_cv = cv2.imread(line.strip()) 
+        img = cv2.cvtColor(im_cv, cv2.COLOR_BGR2RGB)
+        print(img.shape)
+        inputs.append(grpcclient.InferInput(args['input_name'], img.shape, "UINT8"))
+        outputs = []
+        outputs.append(grpcclient.InferRequestedOutput(output_name))
+        
+        #nmpy = np.array(img , dtype=np.uint8)
```
Comment (michalkulakowski) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:94]: remove

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        start_time = datetime.datetime.now()
+        results = triton_client.infer(model_name=graph_name,
+                                  inputs=inputs,
+                                  outputs=outputs)
+        end_time = datetime.datetime.now()
+        duration = (end_time - start_time).total_seconds() * 1000
+        processing_times = np.append(processing_times,np.array([int(duration)]))
+        output = results.as_numpy(output_name)
+        nu = np.array(output)
+        # for object classification models show imagenet class
```
Comment (michalkulakowski) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:105]: remove

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py (hunk 1 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py]: Done.

File: demos/mediapipe/holistic_tracking/holistic_tracking.pbtxt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# dropped, limiting the number of in-flight images in most part of the graph to
+# 1. This prevents the downstream nodes from queuing up incoming images and data
+# excessively, which leads to increased latency and memory usage, unwanted in
+# real-time mobile applications. It also eliminates unnecessarily computation,
+# e.g., the output produced by a node may get dropped downstream if the
+# subsequent nodes are still busy processing previous inputs.
+
+node {
+  calculator: "HolisticLandmarkCpu"
+  input_stream: "IMAGE:throttled_input_video"
```
Comment (atobiszei) [inline demos/mediapipe/holistic_tracking/holistic_tracking.pbtxt:22]: This is not throttled now

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
+#
+# Copyright (c) 2022 Intel Corporation
```
Comment (atobiszei) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:2]: 2023

File: src/test/mediapipeflow_test.cpp [hunk 5]
```diff
@@ -122,7 +122,7 @@ TEST_F(MediapipeTFTest, Passthrough) {
     size_t dummysInTheGraph = 0;
     checkDummyResponse("out", requestData, request, response, dummysInTheGraph, 1, modelName);
 }
-TEST_F(MediapipeTFTest, DummyInfer) {
+TEST_F(MediapipeTFTest, DISABLED_DummyInfer) {
```
Comment (atobiszei) [inline src/test/mediapipeflow_test.cpp:125]: Rebase change and check if you can enable it back.

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+       # 'face_landmark/face_landmark_with_attention.tflite',
+        'hand_landmark/hand_landmark_full.tflite',
+       # Using full
+       # 'hand_landmark/hand_landmark_lite.tflite',
+        'holistic_landmark/hand_recrop.tflite',
+        'iris_landmark/iris_landmark.tflite',
+        'palm_detection/palm_detection_full.tflite',
+       # Using full
+       # 'palm_detection/palm_detection_lite.tflite',
+       # Need to use OV version
```
Comment (atobiszei) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:92]:
```suggestion
       # Need to use OV version
```
Here If I understood correctly we converted tflite model to TF and then back to tflite and we still use tflite model and not OV IR representation?

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        'pose_detection/pose_detection.tflite',
+        'pose_landmark/pose_landmark_full.tflite',
+       # Not working
+       # 'selfie_segmentation/selfie_segmentation.tflite',
+       # 'selfie_segmentation/selfie_segmentation_landscape.tflite',
```
Comment (atobiszei) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:97]: Those are not used anyway in holistic demo

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    for elem in external_files:
+      sys.stderr.write('downloading file: %s\n' % elem)
+      download_model(elem)
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Sends requests via KServe gRPC API using images in format supported by OpenCV. '
+                                                 'It displays performance statistics and optionally the model accuracy')
+    parser.add_argument('--images_list', required=False, default='input_images.txt', help='path to a file with a list of labeled images')
+    parser.add_argument('--grpc_address',required=False, default='localhost',  help='Specify url to grpc service. default:localhost')
+    parser.add_argument('--grpc_port',required=False, default=9022, help='Specify port to grpc service. default: 9022')
```
Comment (atobiszei) [inline demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:108]: i think that in every different demo we use default 9000

**Review by atobiszei — APPROVED** [review]

File: demos/mediapipe/holistic_tracking/holistic_tracking.pbtxt (hunk 3 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/holistic_tracking.pbtxt:22]: Fixed

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py (hunk 4 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:2]: Done.

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+                                    ' from Google Cloud Storage.')
+        shutil.copyfileobj(response, out_file)
+
+def prepare_models():    
+    external_files = [
+       # Using short range
+       # 'face_detection/face_detection_full_range_sparse.tflite',
+        'face_detection/face_detection_short_range.tflite',
+        'face_landmark/face_landmark.tflite',
+       # Model loading error
```
[the remark this answers is not in the document]
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:82]: Done.

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py (hunk 6 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:92]: Yes

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py (hunk 7 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:97]: Ok

File: demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py (hunk 8 above)
Comment (rasapala) [reply demos/mediapipe/holistic_tracking/mediapipe_holistic_tracking.py:108]: Thats why i used 9022 in my demo :)

File: WORKSPACE [hunk 10]
```diff
@@ -89,7 +89,7 @@ http_archive(
 git_repository(
     name = "mediapipe",
     remote = "https://github.com/openvinotoolkit/mediapipe",
-    commit = "f60c6432636d42e100c7d7bbcb6bdd7b57ebe6c8", # Update calculators from ovms repo
+    commit = "9744c4d402f39ffb0436c27c40d40a0ec0817b8f", # Holistic tracking fix cyclic branch
```
[the remark this answers is not in the document]
Comment (rasapala) [reply WORKSPACE:92]: Ok

File: demos/mediapipe/holistic_tracking/build_dependencies.sh [hunk 11]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (jasiu86) [inline demos/mediapipe/holistic_tracking/build_dependencies.sh:1]: Please add copyright header

Comment (rasapala) [reply demos/mediapipe/holistic_tracking/build_dependencies.sh:1]: Done.

**Review by michalkulakowski — APPROVED** [review]

**Pull request merged**